.github/workflows/First.yml

```diff
@@ -9,8 +9,17 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - name: Build the app
-        run: g++ -std=c++17 *.cpp -o a.out
+        run: ./buildcheck.sh
+      - name: Install valgrind
+        run: sudo apt-get install -y valgrind cppcheck
+      - name: Check memory leaks
+        run: valgrind --leak-check=full --error-exitcode=1 ./a.out test/units/unit1.json test/units/unit2.json
       - name: Generate outputs from units
         run: ./test/generate_outputs.sh
       - name: Check if generated outputs are good
-        run: ./test/test_outputs.sh
\ No newline at end of file
+        run: diff test/outputs.txt test/good_outputs.txt
+      - name: Run style and performance check
+        run: cppcheck *.cpp --enable=performance,style --output-file=performance_and_style_report.txt
+      - uses: actions/upload-artifact@v2
+        with:
+          path: ./performance_and_style_report.txt
```

Comment:
Félig jó, de az a része hiányzik, hogy ha a cppcheck errort vagy warningot jelez, akkor ne kapjon pipát a workflow. (Meg talán érdemes lenne a kódanalízissel kezdeni, hogy ha már az hibát talál, akkor ne is futtassa a programot a többi teszthez.)

Reply:
Ezt bettem a yaml fáljba ezt így már elvileg kilép workflow ha hibát talál a cppcheck `cppcheck *.cpp --enable=warning --error-exitcode=1`